test/unit/lib/plugins/aws/package/compile/events/stream.test.js

```diff
@@ -1441,475 +1444,623 @@ describe('AwsCompileStreamEvents', () => {
       ).to.have.any.keys('FirstEventSourceMappingKinesisSomelongname');
     });
   });
-});
 
-describe('test/unit/lib/plugins/aws/package/compile/events/stream.test.js', () => {
-  describe('regular', () => {
-    let eventSourceMappingResource;
-
-    before(async () => {
-      const { awsNaming, cfTemplate } = await runServerless({
-        fixture: 'function',
-        configExt: {
-          functions: {
-            basic: {
-              events: [
-                {
-                  stream: {
-                    arn: 'arn:aws:kinesis:us-east-1:123456789012:stream/some-long-name',
-                    functionResponseType: 'ReportBatchItemFailures',
-                    tumblingWindowInSeconds: 30,
-                    filterPatterns: [{ eventName: ['INSERT'] }, { eventName: ['MODIFY'] }],
-                    parallelizationFactor: 10,
-                    bisectBatchOnFunctionError: true,
-                    consumer: true,
-                  },
-                },
-              ],
-            },
-            dynamo: {
-              handler: 'basic.handler',
-              events: [
-                {
-                  stream: {
-                    arn: 'arn:aws:dynamodb:region:account:table/foo/stream/1',
-                    batchSize: 1,
-                    startingPosition: 'LATEST',
-                    enabled: false,
-                    batchWindow: 15,
-                    maximumRetryAttempts: 4,
-                    maximumRecordAgeInSeconds: 120,
-                    destinations: {
-                      onFailure: 'arn:aws:sns:region:account:snstopic',
+  describe.only('test/unit/lib/plugins/aws/package/compile/events/stream.test.js', () => {
+    describe('regular', () => {
+      let eventSourceMappingResource;
+      let eventSourceMappingResourceDynamoDB;
+      let arnCfGetAttEventSourceMappingResource;
+      let arnCfJoinEventSourceMappingResource;
+      let arnCfRefEventSourceMappingResourceDynamoDB;
+      let arnCfRefEventSourceMappingResourceKinesis;
+      let destinationGetAttSourceMappingResource;
+      let destinationRefSourceMappingResourceBuzz;
+      let destinationRefSourceMappingResourceFizz;
+      let iamRoleLambdaExecution;
```

Comment:
Let's define and resolve all those variables, directly in tests they're used. It'll make it easier to read and maintain.

(If I see correctly they're not reused among tests)

Reply:
Sure, but I have some concerns I think is important to address.
Half of those variables are used in more than one test. 
I also looked at other tests that were refactored and they had it the same way as in this refactor, with all the variables declared prior to `before` block, event though some of those variables were used only in a single test.

Here is one example in `sqs.tes.js`
https://github.com/serverless/serverless/blob/69b22b2cb83a6e15700c2e64f06370681e1022fb/test/unit/lib/plugins/aws/package/compile/events/sqs.test.js#L8-L13

Most importantly, if I do this then `awsNaming` and `cfTemplate` variables will be out of scope (which would require more refactoring).

If you have a specific working edit in mind of how I can define and resolve those variables without overcomplicating it, I would really appreciate it.

Reply:
`awsNaming` and `cfTemplate` can also be predefined in _describe body.

Let's then keep here only those which are reused in multiple tests. Otherwise, if var is used in a single test, let's keep it defined and resolved just in its context. I think it'll improve readability